views/pages/ConfigurationsPage/index.jsx

```diff
@@ -0,0 +1,164 @@
+import React, { useState } from 'react'
+import { Box, Button } from '@mui/material'
+import { Add } from '@mui/icons-material'
+import { useOutletContext, Link } from 'react-router-dom'
+
+import ConfigurationsTable from '../../tables/ConfigurationsTable'
+import PageHeader from '../../components/PageHeader'
+import { UsersModel } from '../../models'
+import api from '../../api'
+import ShareConfigurationForm from '../../forms/ShareConfigurationForm'
+import { routes } from '../../routes/routes'
+
+const ConfigurationsPage = () => {
+  const {
+    configurations,
+    navigate,
+    setNotification,
+    setConfigurations,
+    setUser,
+    user,
+    users,
+  } = useOutletContext()
+  const { uid, preferences } = user
+  const [open, setOpen] = useState(false)
+  const [options, setOptions] = useState([])
+  const [formValues, setFormValues] = useState([])
+  const [currentConfigId, setCurrentConfigId] = useState('')
+
+  const handleNotification = (message) =>
+    setNotification({ open: true, message })
+  const onSubmit = async (data) => {
+    try {
+      const configAttributes = {
+        readers: UsersModel.createReadersList(data.readers),
+      }
+      await api.userConfigurations.update(
+        uid,
+        currentConfigId,
+        configAttributes
+      )
+
+      await loadAllConfigurations(uid)
+
+      closeForm()
+    } catch (error) {
+      handleNotification(error.message)
+    }
+  }
+  const loadAllConfigurations = async (userId) => {
+    try {
+      const configurations = await api.userConfigurations.all(userId)
+
+      setConfigurations(configurations)
+    } catch (error) {
+      handleNotification(error.message)
+    }
+  }
+  const closeForm = () => {
+    setCurrentConfigId('')
+    setOpen(false)
+    setOptions([])
+    reset()
+  }
+  const handleFormModal = async (config) => {
+    const { owner, readers, _id } = config
+    const options = UsersModel.createUserFriendList(users, owner)
+    const values = UsersModel.createOptionsFromReaders(readers, owner)
+
+    setCurrentConfigId(_id)
+    setFormValues(values)
+    setOptions(options)
+    setOpen(true)
```

Comment:
if we only set the config in state then can we get the options and values from that config instead of setting more state values?

Reply:
Looking for clarity, the dropdown options are the same for everyone, the list of all users where the owner of the chart can share to, the form values is what changes, is this what you mean..?

Reply:
so I'm thinking that instead of `setCurrentConfigId(_id)` we could do `setConfig(config)`. Then we can use constants for the rest of these values.

```js
const formValues = config?.readers && config?.owner
  ? UsersModel.createOptionsFromReaders(config.readers, config.owner)
  : []
const friendOptions = config?.readers && config?.owner
  ? UsersModel.createUserFriendList(config.readers, config.owner)
  : []
```

That was we don't need to manage so much state.

Reply:
we probably don't need to have the options in state and it can be a constant that's initialized at a higher level, the function takes the configuration as a parameter so we don't have to set the config to state either

Reply:
hmm nevermind we have to have options in state to set the value that should not be removed from the list

Reply:
[3f060e8](https://github.com/AMP-SCZ/dpdash/pull/487/commits/3f060e8279d72eb848b54d8b7eb594892d7449a8)